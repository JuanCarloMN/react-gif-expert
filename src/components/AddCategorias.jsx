import { useState } from 'react';
import { PropTypes } from 'prop-types';

export const AddCategorias = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState( '' );

    const onInputChange = ({target}) => {
        setInputValue( target.value);
    }

    const onSubmit = ( event ) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1) return;

        // console.log( parent.categories );
        // setCategories( categories => [...categories, inputValue]);
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label='form'>
            <input 
                type="text"
                placeholder="Buscar Gif"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategorias.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
