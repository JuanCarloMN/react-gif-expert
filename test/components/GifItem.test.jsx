import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {
    const title = 'Titulo';
    const url = 'https://www.explore.com/img/gallery/the-50-most-incredible-landscapes-in-the-whole-entire-world/intro-1672072042.webp';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={ title } url={ url }/> );
        expect( container ).toMatchSnapshot();        
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem title={ title } url={ url }/> );
        // screen.debug();
        // console.log(screen.getByRole('img').alt);
        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').alt).toBe( title );
        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    })
    
    test('Debe de mostrar el titulo en el componente', () => {
        render( <GifItem title={ title } url={ url }/> );
        expect( screen.getByText( title )).toBeTruthy();
    })
    
});
