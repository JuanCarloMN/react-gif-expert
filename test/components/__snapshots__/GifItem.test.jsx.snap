// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://www.explore.com/img/gallery/the-50-most-incredible-landscapes-in-the-whole-entire-world/intro-1672072042.webp"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
